### PR TITLE
Integrate match service with frontend

### LINKED ISSUE #72: Matching Service Authentication

### PR DESCRIPTION
**Pull request by asaierika**
Upon connection to the match service socket, frontend sends token to the match service which then calls user service to verify the token and retrieves the userId. Match service then generates a ticketId and sends it to the frontend.
When matching is successful, a "Match found" notification is shown and it returns to the match selection page. The userIds for matched users are printed in console for debugging purpose and should be deleted later.
When a matching is cancelled or timed out, a "Match not found" notification is shown and it returns to the match selection page.
Match DTOs and message types are moved to the common folder.

TODO: env is not up to date with #67 
Closes #72, #74, #61

### CODE REVIEW

File: common/shared-types/index.ts [hunk 1]
```diff
@@ -65,9 +69,57 @@ export type ExtractedUser = z.infer<typeof extractedUserSchema>;
...
+    userIds: z.string().min(1).array().min(1),
+    ticketIds: z.string().min(1).array().min(1),
+  });
+
+export const userTicketSchema = z.object({
+  ticketId: z.string().min(1),
+  data: userMatchingDataSchema,
+});
+
+export const userTicketPlayloadSchema = z.object({
```
Comment (donljh) [inline common/shared-types/index.ts]: Misspelled payload here

File: services/matching/.env.development [hunk 2]
```diff
@@ -0,0 +1,3 @@
+USER_SERVICE_URL="http://localhost:3002"
```
Comment (donljh) [inline services/matching/.env.development]: Should only commit `.env.*.example`

File: services/user/.env.development [hunk 3]
```diff
@@ -0,0 +1,10 @@
+PORT=3002
```
Comment (donljh) [inline services/user/.env.development]: Here too

File: services/user/.env [hunk 4]
```diff
@@ -1,5 +1,5 @@
 PORT=3002
```
Comment (donljh) [inline services/user/.env:1]: Here as well

File: services/question/.env.development [hunk 5]
```diff
@@ -0,0 +1,6 @@
+GOOGLE_APPLICATION_CREDENTIALS='../../.firebase/service-account.json'
```
Comment (donljh) [inline services/question/.env.development]: Here as well

File: services/user/.env (hunk 4 above)
Comment (asaierika) [reply services/user/.env:1]: Isn't .env supposed to be checked in according to #67?

Comment (donljh) [reply services/user/.env:1]: Ah oops

**Review by donljh — DISMISSED** [review]

**Pull request merged**